mathics/core/expression.py

```diff
@@ -861,6 +864,11 @@ def get_sort_key(self, pattern_sort=False):
                 return [1 if self.is_numeric() else 2, 3, head, self._elements, 1]
 
     def has_changed(self, definitions):
+
+        # Some Atoms just don't have a cache.
+        if not hasattr(self, "_cache"):
+            return
```

Comment:
It don't need to be a boolean?

Reply:
Yes, it would be better as a boolean so that is fixed now. Python is loose in that `return` is `return None` and `None` behaves like  `False`.  But better to be explicit about the intent.